From c35e88dff8283859b216e13f9cf1b927813da5fe Mon Sep 17 00:00:00 2001
From: Juha Vuolle <juha.vuolle@insta.fi>
Date: Sat, 26 Feb 2022 16:07:52 +0200
Subject: [PATCH 4/5] Handle LE enhanced connection complete in Bluez
 peripheral

Newer bluetooth devices may trigger (only) a
HCI_LE_Enhanced_Connection_Complete instead of
HCI_LE_Connection_Complete.
This commit adds handling of this so we get a proper bluetooth handle.

Pick-to: 5.15 6.2 6.3
Task-number: QTBUG-101309
Change-Id: Ibb5cf8ca063df9345a0ef0bcb12ae0dd780bab78
Reviewed-by: Alex Blasche <alexander.blasche@qt.io>
Reviewed-by: Ivan Solovev <ivan.solovev@qt.io>
(cherry picked from commit 12afa962ebb5a6eac4b598bdd1b1804c9a3ca089)
---
 src/bluetooth/bluez/hcimanager.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/bluetooth/bluez/hcimanager.cpp b/src/bluetooth/bluez/hcimanager.cpp
index e2635fae..a8b8e3b9 100644
--- a/src/bluetooth/bluez/hcimanager.cpp
+++ b/src/bluetooth/bluez/hcimanager.cpp
@@ -563,9 +563,11 @@ void HciManager::handleHciAclPacket(const quint8 *data, int size)
 
 void HciManager::handleLeMetaEvent(const quint8 *data)
 {
-    // Spec v4.2, Vol 2, part E, 7.7.65ff
+    // Spec v5.3, Vol 4, part E, 7.7.65.*
     switch (*data) {
-    case 0x1: {
+    case 0x1: // HCI_LE_Connection_Complete
+    case 0xA: // HCI_LE_Enhanced_Connection_Complete
+    {
         const quint16 handle = bt_get_le16(data + 2);
         emit connectionComplete(handle);
         break;
-- 
2.38.1

